### ISSUE TITLE
Invalid Virtual Address, the IP address already exists

### ISSUE THREAD

**Issue by payalsin** [F5Networks/f5-appsvcs-extension#186]
##### ISSUE TYPE

 - Bug Report

##### AS3 BUILD/ VERSION
```
3.14.0 |4
```

##### BIGIP VERSION

```
12.1.1.00.136
```

##### SUMMARY

```
Duplicate VIP on a different port not configured
```

##### STEPS TO REPRODUCE
- Manually Add a VIP in common partition with VIP 10.10.10.10 listening on Port 80
- POST a AS3 declaration with the same VIP (10.10.10.10) but a different Port (8080) - different partition
- AS3 returns an error

**Note**: Manually if VIP 10.10.10.10:8080 is added to a different partition it works**

```
Request:

{
    "class": "ADC",
    "schemaVersion": "3.12.0",
    "Tenant": {
        "class": "Tenant",
        "Application": {
            "class": "Application",
            "template": "generic",
            "vip": {
                "class": "Service_HTTP",
                "virtualPort": 8080,
                "virtualAddresses": [
                    "10.10.10.10"
                ],
                "iRules": [
                    {
                        "bigip": "/Common/_sys_https_redirect"
                    },
                    "remote_irule"
                ]
            },
            "remote_irule": {
                "class": "iRule",
                "iRule": {
                    "url": "https://raw.githubusercontent.com/f5devcentral/f5-automation-labs/master/files/irules/maintenance-page.irule"
                }
            }
        }
    }
}

Response

{"results":[{"code":200,"message":"no change","host":"localhost","tenant":"Common","runTime":984},{"code":422,"message":"declaration failed","response":"0107176c:3: Invalid Virtual Address, the IP address 10.10.10.10 already exists.","host":"localhost","tenant":"Tenant","runTime":1361},{"code":200,"message":"no change","host":"localhost","tenant":"Common","runTime":1099}],"declaration":{"class":"ADC","schemaVersion":"3.12.0","id":"autogen_3b5ae42a-02d3-48eb-990c-457c4bed87ab","updateMode":"selective","controls":{"archiveTimestamp":"2019-11-13T18:38:26.448Z"}},"code":207}

```

**Comment by focrensh** [comment]
This is expected as the VirtualAddress already belongs to the Common Tenant. The new "Tenant" cannot own it due to this.

Side note, the 2 irules in the declaration are taking conflicting action (redirect and returning content).

**Comment by dstokesf5** [comment]
As @focrensh already hinted at, this is a TMSH restriction and not an AS3 restriction.

**Comment by dstokesf5** [comment]
Closing for inactivity.

**Comment by thepowercoders** [comment]
I also have this issue but I believe it was closed incorrectly. If I add a VIP to a partition and then add another VIP to a different partition with the same IP address but a different port, the GUI and TMSH allows me to do this. However, with AS3, the API errors.

Example:
![image](https://github.com/F5Networks/f5-appsvcs-extension/assets/32461620/05dd9f10-49af-49ba-a80c-804b2f4cf3b1)

I have added 3 VIPS - one to the Common partition, and 1 each to "test1" and "test2" partitions. The IP address is the same, but the port is unique - this is added with no problems and tests fine.

However, if I try and run an AS3 declaration with the same config, I get the following error:

``` json
 Error: posting as3 config failed for tenants:(dccdm-eit) with error: Tenant Creation failed with Response: [
       {
               "code": 422,
               "host": "localhost",
               "response": "0107176c:3: Invalid Virtual Address, the IP address 192.168.1.1 already exists.",
               "runTime": 22895,
               "tenant": "test1"
       }
 ]
```

The logic here should be that a VIP cannot be added if another partition has the same IP+Port, but if either is different, it should be allowed.
